### PR TITLE
Refactor Button tests to use `render` @testing-library function

### PR DESCRIPTION
**Pull request by Lukazovic**
#### What does this PR do?

Update Button legacy tests to use the standard render method from [react-testing-library](https://testing-library.com/docs/react-testing-library/intro/).

#### Where should the reviewer start?

`src/js/components/Button/__tests__/Button-test.js`

#### What testing has been done on this PR?

`yarn test`

#### How should this be manually tested?

Run: `yarn test`

#### What are the relevant issues?

#5197 - Testing - Refactor the usage of `renderer.create` to `render`

#### Do the grommet docs need to be updated?

No.

#### Should this PR be mentioned in the release notes?

No.

#### Is this change backwards compatible or is it a breaking change?

Backwards compatible.

### CODE REVIEW

**Review by ShimiSun — APPROVED** [review]

File: src/js/components/Button/__tests__/Button-test.js [hunk 1]
```diff
@@ -116,41 +113,39 @@ describe('Button', () => {
...
   test('fill', () => {
-    const component = renderer.create(
+    const { container } = render(
       <Grommet>
-        <Button>
-          <Button fill />
-          <Button fill={false} />
-          <Button fill="horizontal" />
-          <Button fill="vertical" />
-        </Button>
```
Comment (jcfilben) [inline src/js/components/Button/__tests__/Button-test.js:149]: Good catch here, I agree that it doesn't make sense for the 4 Buttons to be wrapped within a Button for this test. I think this is a good change that simplifies the test.

**Review by jcfilben — APPROVED** [review]
Looks good! All the changes look right to me

File: src/js/components/Button/__tests__/Button-test.js (hunk 1 above)
Comment (Lukazovic) [reply src/js/components/Button/__tests__/Button-test.js:149]: Thanks!

I've noticed there are some other Button tests we can improve but I tried not to do it in the same PR to not make it hard to understand.

Comment (jcfilben) [reply src/js/components/Button/__tests__/Button-test.js:149]: Yeah good call not doing those other changes in this PR, but if you think some of the other tests could be improved definitely feel free to file a separate PR with those changes

Comment (Lukazovic) [reply src/js/components/Button/__tests__/Button-test.js:149]: Cool, I'll do it soon

**Comment by ShimiSun** [comment]
Thank you for your contribution! 💟

**Pull request merged**